errors.go

```diff
@@ -89,6 +89,10 @@ const (
 	ErrOffsetsLoadInProgress           KError = 14
 	ErrConsumerCoordinatorNotAvailable KError = 15
 	ErrNotCoordinatorForConsumer       KError = 16
+	ErrInvalidTopic                    KError = 17
+	ErrMessageSetSizeTooLarge          KError = 18
+	ErrNotEnoughReplicas               KError = 19
+	ErrNotEnoughReplicasAfterAppend    KError = 20
```

Comment:
The first is raised if there are not enough ISR when the request is received, so it fails immediately. The second is raised if there are enough initially, but by the time the local append has happened and all remaining ISRs have acked, the number of ISRs is not enough.

Basically: retrying the second could give you duplicates, retrying the first definitely won't.